### PR TITLE
added: home page UI

### PR DESCRIPTION
**Pull request by rkapoor10**
Home Page UI
- Banner Image with CTA
- Video Cards 
- Section for videos under different heading
homeUI ss

### CODE REVIEW

File: src/components/shared/sidemenu/SideMenu.jsx [hunk 1]
```diff
@@ -1,31 +1,45 @@
...
-    { name: "Liked", icon: <AiFillLike /> },
-    { name: "History", icon: <RiHistoryLine /> },
-    { name: "Settings", icon: <RiSettings4Fill /> },
+    { name: "Home", icon: <ImHome />, path: "/home" },
+    { name: "Explore", icon: <MdExplore />, path: "/explore" },
+    { name: "Playlist", icon: <RiPlayList2Fill />, path: "/playlist" },
+    { name: "Watch Later", icon: <MdWatchLater />, path: "/watchlater" },
+    { name: "Liked", icon: <AiFillLike />, path: "/liked" },
+    { name: "History", icon: <RiHistoryLine />, path: "/history" },
+    { name: "Settings", icon: <RiSettings4Fill />, path: "/mockman" },
```
Comment (Bhakti199) [inline src/components/shared/sidemenu/SideMenu.jsx:22]: liked it.

File: src/components/shared/sidemenu/sidemenu.css [hunk 2]
```diff
@@ -1,12 +1,20 @@
 .sidemenu-container{
     padding-top: 1rem;
-    min-height:90vh;
     width:15rem;
+    height:100%;
     background-color: 
     #202020;
```
Comment (Bhakti199) [inline src/components/shared/sidemenu/sidemenu.css:6]: assign color variable

File: src/layout.css [hunk 3]
```diff
@@ -1,27 +1,28 @@
...
-    "side main main main main main"
-    "side main main main main main"
-    "side main main main main main"
-    "foot foot foot foot foot foot";
+    "head head head head head head head head"
+    "side main main main main main main main";
+  /* "side main main main main main main main"
+    "side main main main main main main main"
+    "side main main main main main main main"
+    "foot foot foot foot foot foot foot foot"; */
```
Comment (Bhakti199) [inline src/layout.css:9]: looks cool. UI is looking cool.

**Review by Bhakti199 — APPROVED** [review]
- Code is well structured.
- Jsx is properly maintained and written. 
- Export components from single index file.
- Everthing looks clean.
- dark theme looks cool 😍.
- This PR is good to merge.

**Review by SuyashTiwariCoder — APPROVED** [review]
Overall Code structure is  clean . Nicely done everything .

File: src/components/cards/videoCard.css [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+
+.on-thumbnail-btns:hover .btns-extension{
+    display: inline;
+}
+
+
+
+
+
```
Comment (JantuDeb) [inline src/components/cards/videoCard.css:84]: remove unwanted empty space

File: src/components/shared/sidemenu/SideMenu.jsx (hunk 1 above)
Comment (JantuDeb) [inline src/components/shared/sidemenu/SideMenu.jsx:22]: This is really a good approach

File: src/components/shared/sidemenu/SideMenu.jsx [hunk 5]
```diff
@@ -1,31 +1,45 @@
...
-            {" "}
-            <div className="menu-icon">{icon}</div>{" "}
-            <div className="menu-title">{name}</div>{" "}
-          </li>
+        {menuOptions.map(({ name, icon, path }) => (
+          <NavLink
+            to={path}
+            key={name}
+            className="menu-item"
+            style={({ isActive }) => isActive ?activeStyle:undefined}
```
Comment (JantuDeb) [inline src/components/shared/sidemenu/SideMenu.jsx:38]: You can just use the `active` className which will be added when you use `NavLink`

**Review by JantuDeb — APPROVED** [review]
This looks good, added few comments do check it out

File: src/components/cards/videoCard.css (hunk 4 above)
Comment (rkapoor10) [reply src/components/cards/videoCard.css:84]: done

File: src/components/shared/sidemenu/SideMenu.jsx (hunk 5 above)
Comment (rkapoor10) [reply src/components/shared/sidemenu/SideMenu.jsx:38]: ok

File: src/components/shared/sidemenu/SideMenu.jsx (hunk 1 above)
Comment (rkapoor10) [reply src/components/shared/sidemenu/SideMenu.jsx:22]: Thank You!

File: src/layout.css (hunk 3 above)
Comment (rkapoor10) [reply src/layout.css:9]: Thanks

File: src/components/shared/sidemenu/sidemenu.css (hunk 2 above)
Comment (rkapoor10) [reply src/components/shared/sidemenu/sidemenu.css:6]: okay

File: src/components/shared/sidemenu/SideMenu.jsx (hunk 1 above)
Comment (rkapoor10) [reply src/components/shared/sidemenu/SideMenu.jsx:22]: Thank You!

**Pull request merged**